hedera-node/hedera-app/src/main/java/com/hedera/node/app/throttle/impl/NetworkUtilizationManagerImpl.java

```diff
@@ -0,0 +1,202 @@
+                    e.getMessage()));
+            activeGasThrottle.resetUsageTo(currGasThrottleUsageSnapshot);
+        }
+
+        final var congestionLevelStarts = states.<CongestionLevelStarts>getSingleton(
+                        CongestionThrottleService.CONGESTION_LEVEL_STARTS_STATE_KEY)
+                .get();
+        if (!congestionLevelStarts.genericLevelStarts().isEmpty()) {
+            final var genericLevelStarts = congestionLevelStarts.genericLevelStarts().stream()
+                    .map(ts -> Instant.ofEpochSecond(ts.seconds(), ts.nanos()))
+                    .toArray(Instant[]::new);
+            multiplierSources.resetGenericCongestionLevelStarts(genericLevelStarts);
+        }
+
+        if (!congestionLevelStarts.gasLevelStarts().isEmpty()) {
+            final var gasLevelStarts = congestionLevelStarts.gasLevelStarts().stream()
+                    .map(ts -> Instant.ofEpochSecond(ts.seconds(), ts.nanos()))
+                    .toArray(Instant[]::new);
+            multiplierSources.resetGasCongestionLevelStarts(gasLevelStarts);
+        }
+    }
+
+    @Override
+    public void saveTo(final @NonNull HederaState state) {
+        final var states = state.createWritableStates(CongestionThrottleService.NAME);
+        final var throttleSnapshotsState = states.<ThrottleUsageSnapshots>getSingleton(
+                CongestionThrottleService.THROTTLE_USAGE_SNAPSHOTS_STATE_KEY);
+        final var tpsThrottleUsageSnapshots = handleThrottling.allActiveThrottles().stream()
+                .map(DeterministicThrottle::usageSnapshot)
+                .map(this::toPbj)
+                .toList();
+
+        final var throttleUsageSnapshots = ThrottleUsageSnapshots.newBuilder()
+                .tpsThrottles(tpsThrottleUsageSnapshots)
+                .gasThrottle(toPbj(handleThrottling.gasLimitThrottle().usageSnapshot()))
+                .build();
+
+        throttleSnapshotsState.put(throttleUsageSnapshots);
+
+        final var congestionLevelStartsState =
+                states.<CongestionLevelStarts>getSingleton(CongestionThrottleService.CONGESTION_LEVEL_STARTS_STATE_KEY);
+        final var genericCongestionStarts = Arrays.stream(multiplierSources.genericCongestionStarts())
+                .map(inst -> new Timestamp(inst.getEpochSecond(), inst.getNano()))
+                .toList();
+        final var gasCongestionStarts = Arrays.stream(multiplierSources.gasCongestionStarts())
+                .map(inst -> new Timestamp(inst.getEpochSecond(), inst.getNano()))
+                .toList();
+
+        final var congestionLevelStarts = CongestionLevelStarts.newBuilder()
+                .genericLevelStarts(genericCongestionStarts)
+                .gasLevelStarts(gasCongestionStarts)
+                .build();
+
+        congestionLevelStartsState.put(congestionLevelStarts);
+    }
+
+    private DeterministicThrottle.UsageSnapshot fromPbj(ThrottleUsageSnapshot snapshot) {
```

Comment:
Does it make sense to move these methods to `PbjConverter`?

Reply:
Updated.